### PR TITLE
Use upstream access control headers

### PR DESCRIPTION
**Pull request by mvdbeek**

### CODE REVIEW

File: env/main/templates/nginx/usegalaxy.j2 [hunk 1]
```diff
@@ -141,6 +141,11 @@ server {
...
         add_header X-Frame-Options SAMEORIGIN;
         add_header X-Content-Type-Options nosniff;
+
+        # Add upstream response headers that would otherwise be omitted
+        add_header Access-Control-Allow-Origin $upstream_http_access_control_allow_origin;
+        add_header Access-Control-Allow-Methods $upstream_http_access_control_allow_methods;
+
         # CORS
         #add_header Access-Control-Allow-Origin "http://registry.intermine.org";
         if ($http_origin ~ "^https?://(covid19\.galaxyproject\.org|registry\.intermine\.org)$") {
```
Comment (hexylena) [inline env/main/templates/nginx/usegalaxy.j2:151]: is this still necessary then?

Comment (mvdbeek) [reply env/main/templates/nginx/usegalaxy.j2:151]: Probably not, but there are additional directives I'm not setting ... we can try this on test first.

Comment (mvdbeek) [reply env/main/templates/nginx/usegalaxy.j2:151]: (i mean `Expose-Headers` and `Allow-Headers`)

Comment (hexylena) [reply env/main/templates/nginx/usegalaxy.j2:151]: > we can try this on test first.

sounds good. (i've linked this thread from the related EU playbook change, so, they'll surely check here for the conclusion.)

**Comment by mira-miracoli** [comment]
This should also work with a second proxy (traefik) load-balancing multiple servers running nginx right?

**Comment by mvdbeek** [comment]
As long as the upstream doesn't drop any response headers it should work, yes.

**Review by natefoo — APPROVED** [review]
LGTM and much nicer than what we were doing.

**Comment by sanjaysrikakulam** [comment]
Can anyone confirm that we can remove the `CORS`?

**Comment by sanjaysrikakulam** [comment]
Can anyone please confirm that this works and we can remove the CORS from [our config](https://github.com/usegalaxy-eu/infrastructure-playbook/pull/1157)?

**Comment by mvdbeek** [comment]
How can we tell ? I know avivator works but I don't know why those other headers were set. You can try this yourself and report if anything doesn't work.

**Pull request merged**